cloudinit/config/cc_refresh_rmc_and_interface.py

```diff
@@ -0,0 +1,114 @@
+**Summary:** Ensure Network Manager is not managing IPv6 interface
+
+This module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components
+that together provide a comprehensive clustering environment(RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource
+Monitoring and Control (RMC) subsystem. RMC is a generalized framework used
+for managing, monitoring, and manipulating resources. RMC runs as a daemon
+process on individual machines and needs creation of unique node id and
+restarts during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- Refreshing RMC
+- Disabling NetworkManager from handling IPv6 interface, as IPv6 interface
+  is used for communication between RMC daemon and PowerVM hypervisor.
+
+**Internal name:** per always
+
+**Supported distros:** RHEL
+
+"""
+
+from cloudinit import log as logging
+from cloudinit.settings import PER_ALWAYS
+from cloudinit import util
+from cloudinit import netinfo
+from cloudinit import subp
+
+frequency = PER_ALWAYS
+
+LOG = logging.getLogger(__name__)
+# Ensure that /opt/rsct/bin has been added to standard PATH of the
+# distro. The symlink to rmcctrl is /usr/sbin/rsct/bin/rmcctrl .
+RMCCTRL = 'rmcctrl'
+SED = 'sed'
+
+
+def handle(name, _cfg, _cloud, log, _args):
+    info = netinfo.netdev_info()
+    # Setting NM_CONTROLLED=no for IPv6 interface
+    # making it down and up
+    info_len = len(info.keys())
+
+    LOG.debug('Making the IPv6 up explicitly. '
+              'Ensuring IPv6 interface is not being handled by NetworkManager'
+              ' and it is  restarted to re-establish the communication with '
+              'the hypervisor')
+    for info_count in range(info_len):
+        try:
+            if 'lo' in list(info.keys())[info_count]:
+                LOG.debug('Skipping localhost interface')
+                continue
+            interface = list(info.keys())[info_count]
+            if not info[str(list(info.keys())[info_count])]['ipv4']:
+                disable_NM_ipv6(interface)
+                refresh_ipv6(interface)
+        except BaseException:
+            util.logexc(LOG, 'Interface is not up')
+
+    refresh_rmc()
+
+
+def refresh_ipv6(interface):
+    # IPv6 interface is explicitly brought up, subsequent to which the
+    # RMC services are restarted to re-establish the communication with
+    # the hypervisor.
+    subp.subp(['ip', 'link', 'set', interface, 'down'])
+    subp.subp(['ip', 'link', 'set', interface, 'up'])
+
+
+def disable_NM_ipv6(interface):
+    # Ensuring that the communication b/w the hypervisor and VM is not
+    # interrupted due to NetworkManager. For this purpose, as part of
+    # this function, the NM_CONTROLLED is explicitly set to No for IPV6
+    # interface and NetworkManager is restarted.
+    interface_file = '/etc/sysconfig/network-scripts/ifcfg-' + interface
```

Comment:
A log message would probably be good here.

LOG.debug("Editing interface file '%s' for iface '%s", interface_file, interface)